pkg/cmd/server/server_test.go

```diff
@@ -80,37 +80,51 @@ func TestDefaultVolumeSnapshotLocations(t *testing.T) {
 
 	defaultVolumeSnapshotLocations := make(map[string]string)
 
-	// No defaults
-	volumeSnapshotLocations, err := getDefaultVolumeSnapshotLocations(arkClient, namespace, defaultVolumeSnapshotLocations)
-	assert.Equal(t, 0, len(volumeSnapshotLocations))
+	// No default: a default VSL does not need to be specified if there’s only one vsl for a given provider
+	volumeSnapshotLocations, err := getVolumeSnapshotLocations(arkClient, namespace, defaultVolumeSnapshotLocations)
+	assert.Equal(t, 1, len(volumeSnapshotLocations))
 	assert.NoError(t, err)
 
-	// Bad location
-	defaultVolumeSnapshotLocations["provider1"] = "badlocation"
-	volumeSnapshotLocations, err = getDefaultVolumeSnapshotLocations(arkClient, namespace, defaultVolumeSnapshotLocations)
+	// 1 default
+	defaultVolumeSnapshotLocations["provider1"] = "location1"
+	volumeSnapshotLocations, err = getVolumeSnapshotLocations(arkClient, namespace, defaultVolumeSnapshotLocations)
+	assert.Equal(t, 1, len(volumeSnapshotLocations))
+	assert.NoError(t, err)
+
+	// No default fails when there’s more than one vsl for a that provider
+	defaultVolumeSnapshotLocations = make(map[string]string)
+	location3 := &v1.VolumeSnapshotLocation{ObjectMeta: metav1.ObjectMeta{Name: "location3"}, Spec: v1.VolumeSnapshotLocationSpec{Provider: "provider1"}}
+	arkClient.ArkV1().VolumeSnapshotLocations(namespace).Create(location3)
+	volumeSnapshotLocations, err = getVolumeSnapshotLocations(arkClient, namespace, defaultVolumeSnapshotLocations)
 	assert.Equal(t, 0, len(volumeSnapshotLocations))
 	assert.Error(t, err)
 
-	// Bad provider
+	// Non-existing provider given in a < 2 vsl scenario
+	defaultVolumeSnapshotLocations["provider1"] = "badlocation"
+	volumeSnapshotLocations, err = getVolumeSnapshotLocations(arkClient, namespace, defaultVolumeSnapshotLocations)
+	assert.Equal(t, 1, len(volumeSnapshotLocations))
```

Comment:
A comment here as to what the 1 location return should be is helpful. I had to go to the beginning of the function to find that out.